sdk/src/client/api/block_builder/input_selection/requirement/amount.rs

```diff
@@ -247,13 +247,17 @@ impl InputSelection {
 
     fn reduce_funds_of_chains(&mut self, amount_selection: &mut AmountSelection) -> Result<(), Error> {
         // Only consider automatically transitioned outputs.
-        let outputs = self.outputs.iter_mut().filter(|output| {
-            output
-                .chain_id()
-                .as_ref()
-                .map(|chain_id| self.automatically_transitioned.contains(chain_id))
-                .unwrap_or(false)
-        });
+        let outputs = self
+            .provided_outputs
```

Comment:
We could remove if we want to loop a bit less? automatically transitioned can never be part of provided?

Reply:
good point!